### PR TITLE
Update federation controllers to use release 1.5 clientset

### PR DESCRIPTION
**Pull request by caesarxuchao**
This PR is based on #32243, we won't merge it until next week, to make cherry-picks easier.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32246)

### CODE REVIEW

**Comment by nikhiljindal** [comment]
cc @kubernetes/sig-cluster-federation

**Comment by caesarxuchao** [comment]
@nikhiljindal Rebased, PTAL. Thanks!

File: federation/pkg/federation-controller/namespace/namespace_controller.go [hunk 1]
```diff
@@ -22,14 +22,14 @@ import (
 	"time"
 
 	federation_api "k8s.io/kubernetes/federation/apis/federation/v1beta1"
-	federation_release_1_4 "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	federation_kubeclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_5"
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/namespace/namespace_controller.go]: same

File: federation/pkg/federation-controller/ingress/ingress_controller.go [hunk 2]
```diff
@@ -23,14 +23,14 @@ import (
 	"time"
 
 	federation_api "k8s.io/kubernetes/federation/apis/federation/v1beta1"
-	federation_release_1_4 "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	federation_kubeclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_5"
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/ingress/ingress_controller.go]: At other places, we use federationclientset. Should be consistent.

**Comment by nikhiljindal** [comment]
LGTM, thx!

File: federation/pkg/federation-controller/namespace/namespace_controller.go (hunk 1 above)
Comment (caesarxuchao) [inline federation/pkg/federation-controller/namespace/namespace_controller.go]: I'll rename it. Thanks.

File: pkg/client/clientset_generated/release_1_5/typed/core/v1/fake/fake_pod_expansion.go [hunk 3]
```diff
@@ -44,3 +45,14 @@ func (c *FakePods) GetLogs(name string, opts *v1.PodLogOptions) *restclient.Requ
 	_, _ = c.Fake.Invokes(action, &v1.Pod{})
 	return &restclient.Request{}
 }
+
+func (c *FakePods) Evict(eviction *policy.Eviction) error {
```
Comment (caesarxuchao) [inline pkg/client/clientset_generated/release_1_5/typed/core/v1/fake/fake_pod_expansion.go]: This is to fix the unit test failure. The function is required by the PodInterface. @nikhiljindal

**Comment by caesarxuchao** [comment]
I rebased the code, addressed the comment on the inconsistent import names, fixed a build failure. They are all quite trivial, I added back the lgtm myself, @nikhiljindal please remove it if you see any issue. Thanks!

**Comment by caesarxuchao** [comment]
@k8s-bot test this, issue #IGNORE

**Comment by caesarxuchao** [comment]
@k8s-bot gke e2e test this, issue #33611

**Pull request merged**